### ISSUE TITLE
Do not allow "since dates" in the future

### ISSUE THREAD

**Issue by simonrozsival** [toggl/mobileapp#1592]
This problem can happen for example when user changes his/her device time manually (to the future) and creates some entities (e.g., a tag) - when this entity is synced, our storage will save the date time and further pulling will be impossible, because user will be caught in an infinite retry loop (API returns Bad Request).

**Comment by amulware** [comment]
I don't understand how this can happen. The since date used should always be an AT date returned by backend, which can never be in the future unless backend derps. No?

**Comment by amulware** [comment]
@simonrozsival Please elaborate here or explain internally.

**Comment by simonrozsival** [comment]
Steps to reproduce:
- change your phone date to some random date in the future (‘X’)
- sync the app
- change the phone settings to current date
- the app won’t pull anything until ‘X’

**Comment by simonrozsival** [comment]
This has to wait until #1993 is merged.

**Comment by simonrozsival** [comment]
I can't remember how I got into a state where the stored since param was in the future. I doesn't make sense to me anymore... I know there was a problem, but I now think it had a different cause. I think we should close this until we come across this problem again and are able to reproduce it consistently.

**Comment by amulware** [comment]
Was this related to having nullable At dates that were set to 'now' as default?

**Comment by simonrozsival** [comment]
You are right, that could be the problem. I will give this more thought - but I am quite sure this will not be an issue anymore (definitely not when #1993 is merged).

**Comment by amulware** [comment]
I also think that it should be fine with #1993. But have a think, and if you're sure this cannot happen anymore, feel free to close.

**Comment by simonrozsival** [comment]
Well. We still have the default value for the `At` property for tags (see https://github.com/toggl/mobileapp/blob/develop-syncing-refactoring/Toggl.Ultrawave/Models/Tag.cs#L26) - so the question is, what should be the default value instead? Or do we still need this weird constructor?

I will test if not getting the - `At` back after creating the tag it is still a thing.

**Comment by amulware** [comment]
Eek. In any case, if we need fake at-dates, I insist we use a hardcoded constant (shared between all places that use it for easy 'find usages' search) sometime in the past. I see nothing that can go wrong with doing that.

**Comment by simonrozsival** [comment]
We don't need that fake thing anymore. I consistently get the `at` back in the JSON. I will 🔥 that code. If we run into problems during testing (we won't) we will either force backend to fix it or we will use the fixed date as you suggested.

**Comment by amulware** [comment]
Ok. Let's make sure we test this properly.
